### PR TITLE
Bug 2002238: persist imagestream info when switching from yaml to form editor

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes:**
https://issues.redhat.com/browse/ODC-6281

**Root analysis:**
When the namespace is changed then the initialValues for the imagestream and the tag is set to empty string, which is not updated when the user selects new values. As a result, when the user toggles from form editor -> yaml editor -> form editor the values do not persist as it is read from the initial values.

A similar thing happens when the image stream is changed.

**Solution description:**
- removed the usage of initial values from ImageStreamDropdown and ImageStreamNsDropdown

**GIF:**
imagestreampersis

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx [hunk 1]
```diff
@@ -14,49 +14,31 @@ const ImageStreamNsDropdown: React.FC<{ disabled?: boolean; formContextField?: s
...
+    },
+    [dispatch, fieldPrefix, initialIsi, setFieldValue],
+  );
 
   React.useEffect(() => {
-    if (initialImageStream.image && imageStream.image !== initialImageStream.image) {
-      initialImageStream.image = imageStream.image;
-    }
     imageStream.namespace && onDropdownChange();
     // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (rottencandy) [inline frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:34]: Nit: This comment can be removed now.

File: frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx [hunk 2]
```diff
@@ -14,49 +14,31 @@ const ImageStreamNsDropdown: React.FC<{ disabled?: boolean; formContextField?: s
...
-    initialImageStream.image,
-    initialImageStream.tag,
-  ]);
-
   return (
     <ResourceDropdownField
       name={`${fieldPrefix}imageStream.namespace`}
       label={t('devconsole~Project')}
-      title={t('devconsole~Select Project')}
+      title={imageStream.namespace || t('devconsole~Select Project')}
```
Comment (rottencandy) [inline frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:41]: Why is this change needed?

Comment (debsmita1) [reply frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:40]: On page refresh, the namespace selector dropdown shows `Select project` instead of the selected namespace

**Review by rottencandy — APPROVED** [review]
Tested changes locally.
/lgtm
